### PR TITLE
Quick fix for the charm browser URLs for release.

### PR DESCRIPTION
**Pull request by frankban**
Fixes https://github.com/juju/juju-gui/issues/2401

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6706/

**Comment by frankban** [comment]
**QA** must be done in both gijoe and gisf scenarios. 
For gisf, at this stage you can use `guiproxy -config 'gisf: true, jujuEnvUUID: ""'`.
Search, open details in the charm browser, and add to the canvas, as many charms and bundles as you can, including mediawiki (also with series) and kibana. Ensure it works and the URL for the charm/bundle detail page is the new style one (i.e. `[u/{user}/]{name}[/{series}/{revision}]`).

File: jujugui/static/gui/src/app/components/entity-details/entity-details.js [hunk 1]
```diff
@@ -202,15 +202,7 @@ YUI.add('entity-details', function() {
       this.refs.content.focus();
       // Be sure to convert the id to the legacy id as the URL will be in the
       // new id format.
-      const entityId = this.props.id;
-      let processedId = entityId;
-      // If the entityId contains a / and no ~ then it needs to be converted
-      // into the legacy charm format until the charmstore can accept the new
-      // format.
-      if (entityId.indexOf('/') !== -1 && entityId.indexOf('~') === -1) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-details/entity-details.js]: Thanks for removing this hack :)

File: jujugui/static/gui/src/app/components/search-results/item/item.js [hunk 2]
```diff
@@ -143,13 +143,20 @@ YUI.add('search-results-item', function(Y) {
...
-    _handleItemClick: function(id, e) {
-      e.stopPropagation();
+    _handleItemClick: function(id, evt) {
+      evt.stopPropagation();
+      // TODO frankban: it should be clear whether this id is legacy or not.
+      let url;
+      try {
+        url = window.jujulib.URL.fromLegacyString(id);
+      } catch(_) {
+        url = window.jujulib.URL.fromString(id);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results/item/item.js]: The legacy/new urls are going to be fun to untangle sometime in the future.

**Comment by Lukewh** [comment]
QA OK

**Comment by frankban** [comment]
Thanks Jeff and Luke!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge/2098

**Comment by frankban** [comment]
Trying again.
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**